.github/workflows/component-library-ci.yml

```diff
@@ -0,0 +1,121 @@
+name: Component-Library-CI
+
+on:
+  workflow_call:
+  pull_request:
+    branches:
+      - staging
+      - staging-next
+    paths:
+        - 'frontend/packages/component-library/**'
+        - 'frontend/packages/component-library-styles/**'
+        - 'shared/**'
+
+jobs:
+    build:
+        runs-on: ubuntu-24.04
+
+        steps:
+            - uses: actions/checkout@v4
+              with:
+                sparse-checkout: |
+                  frontend
+                  shared/fonts
+                  .github
+
+            - name: Setup Frontend
+              uses: ./.github/actions/frontend/setup
+
+            - name: Install Playwright for Design System Storybook
+              run: npx playwright install --with-deps
+              working-directory: ./frontend/apps/design-system-storybook
+
+            - name: Lint
+              run: yarn lint --filter @code-dot-org/component-library --filter @code-dot-org/design-system-storybook
+              working-directory: ./frontend
```

Comment:
you can set this as a default for the job (or the whole workflow) and then just override when you want to run something from the root or elsewhere.

https://docs.github.com/en/actions/writing-workflows/workflow-syntax-for-github-actions#jobsjob_idstepsworking-directory